### PR TITLE
Fix/cl vault unused funds

### PR DESCRIPTION
**Pull request by njerschow**
## 1. Overview

Changing the CL Vault contract to also use any funds that aren't being saved as rewards

## 2. Implementation details

Added a function that gets the usable balance of the contract

## 4. Checklist

- [ ] Does the Readme need to be updated?

### CODE REVIEW

File: smart-contracts/contracts/cl-vault/src/helpers.rs [hunk 1]
```diff
@@ -244,6 +245,31 @@ pub fn sort_tokens(tokens: Vec<Coin>) -> Vec<Coin> {
     sorted_tokens
 }
 
+/// this function subtracts out anything from the raw contract balance that isn't dedicated towards user or strategist rewards.
+/// this function is expensive.
+pub fn get_unused_balances(
```
Comment (LaurensKubat) [inline smart-contracts/contracts/cl-vault/src/helpers.rs:250]: This needs some tests

File: smart-contracts/contracts/cl-vault/src/rewards/helpers.rs [hunk 2]
```diff
@@ -114,7 +114,18 @@ impl Rewards {
     }
 
     pub fn from_coins(coins: Vec<Coin>) -> Self {
-        Rewards(coins)
+        CoinList(coins)
+    }
+
+    pub fn find_coin(&self, denom: String) -> Coin {
```
Comment (LaurensKubat) [inline smart-contracts/contracts/cl-vault/src/rewards/helpers.rs:120]: the `String` should be an `&str` here

File: smart-contracts/contracts/cl-vault/src/vault/range.rs [hunk 3]
```diff
@@ -134,10 +135,16 @@ pub fn handle_withdraw_position_reply(
...
     let pool_config = POOL_CONFIG.load(deps.storage)?;
-    // what about funds sent to the vault via banksend, what about airdrops/other ways this would not be the total deposited balance
-    // todo: Test that one-sided withdraw wouldn't error here (it shouldn't)
-    let amount0: Uint128 = msg.amount0.parse()?;
-    let amount1: Uint128 = msg.amount1.parse()?;
+
+    let mut amount0: Uint128 = msg.amount0.parse()?;
+    let mut amount1: Uint128 = msg.amount1.parse()?;
+
+    let unused_balances = get_unused_balances(deps.storage, &deps.querier, &env)?;
```
Comment (LaurensKubat) [inline smart-contracts/contracts/cl-vault/src/vault/range.rs:142]: This needs to be tested, can be done by adding free balance to `test_execute_update_range`

**Review by LaurensKubat — CHANGES_REQUESTED** [review]
needs some minor changes then lgtm

**Review by LaurensKubat — DISMISSED** [review]
lgtm

**Review by LaurensKubat — APPROVED** [review]
lgtm

**Review by magiodev — APPROVED** [review]
lgtm

**Pull request merged**